torchaudio/datasets/vctk_092.py

```diff
@@ -0,0 +1,150 @@
+                   folder_txt: str,
+                   downsample: bool = False) -> Tuple[Tensor, int, str, str, str]:
+    speaker_id, utterance_id, _ = fileid.split("_")
+
+    # Read text
+    #
+    # After 30th Nov 2018 update, a new mic was used to record some audio files.
+    # In the process all audio file names were changed to include the mic info.
+    #
+    # Old audio filename - p225_001
+    # New audio filename - p225_001_mic1
+    #
+    # So, using string slicing to use the appropriate fileid for text files.
+    file_txt = os.path.join(path, folder_txt, speaker_id, fileid[:-5] + ext_txt)
+
+    with open(file_txt) as file_text:
+        utterance = file_text.readlines()[0]
+
+    # Read flac
+    file_audio = os.path.join(path, folder_audio, speaker_id, fileid + ext_audio)
+    waveform, sample_rate = torchaudio.load(file_audio)
+    if downsample:
+        # TODO Remove this parameter after deprecation
+        F = torchaudio.functional
+        T = torchaudio.transforms
+        # rate
+        sample = T.Resample(sample_rate, 16000, resampling_method='sinc_interpolation')
+        waveform = sample(waveform)
+        # dither
+        waveform = F.dither(waveform, noise_shaping=True)
+
+    return waveform, sample_rate, utterance, speaker_id, utterance_id
+
+
+class VCTK_092(Dataset):
+    """
+    Create a Dataset for VCTK 0.92. Each item is a tuple of the form:
+    (waveform, sample_rate, utterance, speaker_id, utterance_id)
+    Folder `p315` will be ignored due to the non-existent corresponding text files.
+    For more information about the dataset visit: https://datashare.is.ed.ac.uk/handle/10283/3443
+    """
+
+    _folder_txt = "txt"
+    _folder_audio = "wav48_silence_trimmed"
+    _ext_txt = ".txt"
+    _ext_audio = ".flac"
+    _except_folder = "p315"
+
+    def __init__(self,
+                 root: str,
+                 url: str = URL,
+                 folder_in_archive: str = FOLDER_IN_ARCHIVE,
+                 download: bool = False,
+                 downsample: bool = False,
+                 transform: Any = None,
+                 target_transform: Any = None) -> None:
+
+        if downsample:
+            warnings.warn(
+                "In the next version, transforms will not be part of the dataset. "
+                "Please use `downsample=False` to enable this behavior now, "
+                "and suppress this warning."
+            )
+
+        if transform is not None or target_transform is not None:
+            warnings.warn(
+                "In the next version, transforms will not be part of the dataset. "
+                "Please remove the option `transform=True` and "
+                "`target_transform=True` to suppress this warning."
+            )
+
+        self.downsample = downsample
+        self.transform = transform
+        self.target_transform = target_transform
+
+        archive = os.path.basename(url)
+
+        # url contains un-wanted '?sequence=2&isAllowed=y'
```

Comment:
> # url contains un-wanted '?sequence=2&isAllowed=y'
Does this still apply?

Reply:
yes and we need to slice it

Reply:
Where does it come from?

I mean the default value is `https://datashare.is.ed.ac.uk/bitstream/handle/10283/3443/VCTK-Corpus-0.92.zip`, which does not contain any URL option.

Reply:
Oh I see, this is the case for custom URL provided by a user.

If this concerns only for the case where user provides such URL with options, can you update the comment
> custom `url` may contain options like '?sequence=2&isAllowed=y'

Reply:
will do